src/public/modules/forms/base/directives/submit-form.directive.js

```diff
@@ -353,6 +357,7 @@ function submitFormDirective(
        * @param {string?} toastMessage The toast message to display, if any.
        */
       const handleSubmitFailure = (error, toastMessage) => {
+        scope.responseId = ''
```

Comment:
can I check why this is necessary?

Reply:
Oh right. I realised it is not necessary because the thank you page will not render if submission fails. I will remove this and also make the necessary changes for the comments above as well.